src/engine/controls/keycontrol.cpp

```diff
@@ -234,15 +246,8 @@ void KeyControl::updateRate() {
                 m_pPitchAdjust->set(
                         KeyUtils::powerOf2ToSemitoneChange(m_pitchRateInfo.pitchTweakRatio));
             } else { // Unlock and reset to linear pitch (orig. key + pitch fader offset)
-                // If 'current' aka 'not original' key was locked
-                // TODO(ronso0) Why does it depend on keylock mode?
-                if (m_keylockMode->get() == kLockCurrentKey) {
-                    // reset to linear pitch
```

Comment:
it's not that we reset to linear pitch, it's that we allow rate slider changes to affect the musical pitch.  But 0.0 rate adjustment may no longer the original key.